src/runtime-rs/crates/hypervisor/src/ch/inner_device.rs

```diff
@@ -26,7 +26,37 @@ const DEFAULT_DISK_QUEUE_SIZE: u16 = 1024;
 impl CloudHypervisorInner {
     pub(crate) async fn add_device(&mut self, device: DeviceType) -> Result<()> {
         if self.state != VmmState::VmRunning {
-            self.pending_devices.insert(0, device);
+            // If the VM is not running, add the device to the pending list to
+            // be handled later.
+            //
+            // Note that the only device types considered are DeviceType::ShareFs
+            // and DeviceType::Network since:
+            //
+            // - ShareFs (virtiofsd) is only needed in an non-DM and non-TDX scenario
+            //   for the container rootfs.
+            //
+            // - For all other scenarios, the container rootfs is handled by a
```

Comment:
I think we may need to handle rootfs as block device as well, as I am sure we are supporting that case with the golang runtime. This comment will need to be changed to reflect this, of course that could so in as a future change. 
But something to note here.

Reply:
Container rootfs block devices are handled "post VM boot" so that filters through to us just calling `self.handle_add_device(device)` immediately (aka nothing is added to the pending list).

For a VM rootfs block device, we already have all the info we need so although runtime-rs does call `add_device()` pre-boot, we don't actually need that entry as we just use `PmemConfig::try_from(&boot_info)` currently.